src/managementEnforcer.ts

```diff
@@ -260,6 +260,11 @@ export class ManagementEnforcer extends InternalEnforcer {
    * @return succeeds or not.
    */
   public async addNamedPolicy(ptype: string, ...params: string[]): Promise<boolean> {
+    if (this.autoNotifyWatcher) {
+      // error intentionally ignored
```

Comment:
The comment is not necessary. I think it's pretty self-explanatory if you try to notify a watcher but it does not exist.

I am also thinking about whether we should await for notify functions. Sure, if the update does not go through it's kinda bad, but at the same time, it would halt the current adapter from doing modifications to the policy when it might be crucial. Other adapters can still be synced in other ways.

Example case: if we have an adapter that is a mongo database and a watcher that is Redis. We should not block Casbin from doing updates to the policy if Redis is down. Other enforcers can still be synced via adapter connection if need be, but in this case, the adapter and the enforcer would stop working altogether just because Redis is down.

Reply:
@Sefriol you are right, do you think there is any other way so that we don't fall in watcherEx callback loop

Reply:
I think the current implementation is probably fine, but just do not `await` the watcher. If notify fails, then it fails. Blocking the enforcer / adapter from working when watcher is down is more harmful IMO.

Reply:
Other enforcers most likely do still have access to the policy storage even when the watcher is down, so they do have a chance to update their policy tables by other means.

I am not sure if this should be a configuration parameter. Maybe in some cases, people would want the adapter to be blocked when the watcher is not responsive and sometimes the other way around.